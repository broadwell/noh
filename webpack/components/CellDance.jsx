import React from "react";
import PropTypes from "prop-types";

const CellDance = props => (
  <div className={`cell cell--${props.length} cell--dance`}>{props.text}</div>
);

CellDance.propTypes = {
  text: PropTypes.string.isRequired,
  length: PropTypes.number.isRequired
};

export default CellDance;
