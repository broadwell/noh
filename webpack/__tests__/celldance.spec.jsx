import React from "react";
import { shallow } from "enzyme";
import CellDance from "../components/CellDance";

describe("<CellDance>", () => {
  it("renders as expected by default", () => {
    const component = shallow(<CellDance text="Naka no takane" length={3} />);
    expect(component).toMatchSnapshot();
  });
});
