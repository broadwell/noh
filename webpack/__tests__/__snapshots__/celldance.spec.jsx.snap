// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CellDance> renders as expected by default 1`] = `
<div
  className="cell cell--3 cell--dance"
>
  Naka no takane
</div>
`;
